src/android/CameraPreviewFragment.java

```diff
@@ -309,4 +335,94 @@ public void setLocation(Location loc) {
             this.location = loc;
         }
     }
+
+
+    public void switchCameraTo(String device, CameraSwitchedCallback cameraSwitchedCallback) {
+        Handler mainHandler = new Handler(Looper.getMainLooper());
+        mainHandler.post(() -> {
+            ListenableFuture<ProcessCameraProvider> cameraProviderFuture = ProcessCameraProvider.getInstance(getActivity());
+            ProcessCameraProvider cameraProvider = null;
+            try {
+                cameraProvider = cameraProviderFuture.get();
+            } catch (ExecutionException | InterruptedException e) {
+                Log.e(TAG, "Error occurred while trying to obtain the camera provider: " + e.getMessage());
+                e.printStackTrace();
+                cameraSwitchedCallback.onSwitch(false);
+                return;
+            }
+
+            setUpCamera(device,cameraProvider);
+
+            preview.setSurfaceProvider(viewFinder.getSurfaceProvider());
+            cameraSwitchedCallback.onSwitch(true);
+        });
+    }
+    
+    @SuppressLint("RestrictedApi")
+    public void setUpCamera(String captureDevice, ProcessCameraProvider cameraProvider) {
+        CameraSelector cameraSelector;
+        if (captureDevice.equals("ultra-wide-angle")) {
```

Comment:
```suggestion
        if (captureDevice.equals("ultra-wide")) {
```
I think ultra-wide is a more generic name

Reply:
but i think ultra-wide-angle is more descriptive. @dinitri @YushraJewon Your views?